tests/test_application.py

```diff
@@ -201,6 +201,18 @@ async def post_shutdown(application: Application) -> None:
                 post_shutdown=None,
             )
 
+    def test_job_queue(self, bot, app, recwarn):
+        expected_warning = (
+            "No `JobQueue` set up. To use `JobQueue`, you must install PTB via "
+            "`pip install python-telegram-bot[job_queue]`."
+        )
+        assert app.job_queue is app._job_queue
+        application = ApplicationBuilder().token(bot.token).job_queue(None).build()
+        assert application.job_queue is None
+        assert len(recwarn) == 1
+        assert str(recwarn[0].message) == expected_warning
+        assert recwarn[0].filename == __file__, "wrong stacklevel"
```

Comment:
I would add some comment here explaining the intention. Maybe it's just me, but without reading @Bibo-Joshi 's comments in the review I would be pretty confused if I saw this line in the test

Reply:
specifically which line? the last one?

Reply:
Yes, with "wrong stacklevel". That's funny because I chose that line when reviewing, somehow GH isn't showing it because I made no suggested changes :)

Reply:
And then maybe @Bibo-Joshi wouldn't agree that this line deserves a special comment anyway

Reply:
I'm very much fine with that if it's of help for someone :)

Reply:
After reading up on this in [Python docs](https://docs.python.org/3/library/warnings.html#warnings.warn), I'm not sure anymore if a comment is warranted. I don't know what level of Python proficiency we're expecting :)